paimon-format/src/main/java/org/apache/paimon/format/parquet/ParquetSchemaConverter.java

```diff
@@ -197,8 +196,47 @@ private static Type createTimestampWithLogicalType(
 
     private static List<Type> convertToParquetTypes(RowType rowType) {
         List<Type> types = new ArrayList<>(rowType.getFieldCount());
-        for (int i = 0; i < rowType.getFieldCount(); i++) {
-            types.add(convertToParquetType(rowType.getFieldNames().get(i), rowType.getTypeAt(i)));
+        for (DataField field : rowType.getFields()) {
+            Type parquetType = convertToParquetType(field.name(), field.type());
+            Type typeWithId = parquetType.withId(field.id());
+            if (field.type().getTypeRoot() == DataTypeRoot.ARRAY) {
+                GroupType groupType = (GroupType) parquetType;
+                GroupType wrapperType = (GroupType) groupType.getFields().get(0);
+                Type elementTypeWithId =
+                        wrapperType
+                                .getFields()
+                                .get(0)
+                                .withId(SpecialFields.getArrayElementFieldId(field.id()));
+                typeWithId =
+                        ConversionPatterns.listOfElements(
```

Comment:
Can we just use `convertToParquetType(String name, DataType type, Type.Repetition repetition)`?

It is not good to use `ConversionPatterns.listOfElements` twice.

Reply:
We need to set id for element type here. By calling `convertToParquetType` we can only get an array type, whose element type has no id.